File: services/asset-service/src/main/java/com/cybergate/asset/model/Asset.java
package com.cybergate.asset.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "assets", schema = "asset")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Asset {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AssetType assetType;

    @Column(nullable = false)
    private String environment;

    private String owner;

    private String department;

    private String ipAddress;

    private BigDecimal businessValueInr;

    private BigDecimal replacementCostInr;

    private Boolean internetExposed;

    private Integer criticalityScore;

    private String dataSensitivity;

    private BigDecimal annualRevenueImpact;

    private String metadata;

    @Column(nullable = false)
    private OffsetDateTime createdAt;

    private OffsetDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = OffsetDateTime.now();
        updatedAt = OffsetDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = OffsetDateTime.now();
    }
}

File: frontend/src/types/asset.ts
export interface Asset {
  id: string
  name: string
  asset_type: string
  environment: string
  owner: string
  department: string
  ip_address: string | null
  operating_system: string | null
  business_value_inr: number
  replacement_cost_inr: number
  internet_exposed: boolean
  criticality_score: number
  data_sensitivity: string
  annual_revenue_impact: number
  created_at: string
  updated_at: string
}

export interface AssetDependency {
  id: string
  asset_id: string
  depends_on_id: string
  dependency_type: string
  criticality: number
}

export type AssetType = 'SERVER' | 'DATABASE' | 'APPLICATION' | 'NETWORK' | 'CLOUD' | 'ENDPOINT'
export type Environment = 'PRODUCTION' | 'STAGING' | 'DEVELOPMENT'
export type DataSensitivity = 'PUBLIC' | 'INTERNAL' | 'CONFIDENTIAL' | 'RESTRICTED'
